azure-devops/azext_devops/dev/team/wiki.py

```diff
@@ -33,10 +32,10 @@ def create_wiki(name, wiki_type='projectwiki', mapped_path=None, version=None,
     :param repository: Name or ID of the repository to publish the wiki from. Only required for codewiki type.
     :type repository: str
     """
-    organization, project, repository = resolve_instance_project_and_repo(detect=detect,
-                                                                          organization=organization,
-                                                                          project=project,
-                                                                          repo=repository)
+    organization, project = resolve_instance_and_project(detect=detect,
+                                                         organization=organization,
+                                                         project=project,
+                                                         project_required=True)
     wiki_client = get_wiki_client(organization)
```

Comment:
can we instead make to choice to detect repo or not detect repo based on WikiType?
from a user perspective using --detect I think the expectation should remain same across commands...